lib/logstash/outputs/google_cloud_storage.rb

```diff
@@ -154,6 +155,12 @@ class LogStash::Outputs::GoogleCloudStorage < LogStash::Outputs::Base
 
   config :max_concurrent_uploads, :validate  => :number, :default => 5
 
+  # When true, uploaded files will be placed into sub-folders based on their creation dates.
+  # log_file_prefix configuration will be used as the parent folder and
+  # uploaded files will be placed on sub-folders in dt=%Y-%m-%d format.
+  # Foldered files can be used for creating hive partitioned external tables at BigQuery.
+  config :hive_partition_files, :validate  => :boolean, :default => false
```

Comment:
Instead boolean value (which is very hard to extend in the future), we may directly use the format.
Something like `hive_partition_format`? defaults to no-value, which keeps current behavior.

Reply:
Makes so much sense. My approach was completely static. In this way we can supply different partitioning structures easily. I changed the configuration name and partitioning logic as you suggested.